In [10]:
import pandas as pd

df = pd.read_csv('csv_files/response_clean.csv')
df .head()

,timestamp,year,program,beh_daily_use,beh_time_spent,beh_attendance,hours_per_week_raw,pu_useful,pu_efficiency,il_enjoyment,...,srl_strategy,srl_organize,srl_review,srl_persist,srl_evaluate,open_frustration,open_feature_request,open_srs_usage,hours_per_week_num,hours_bucket
0,9/6/2026 14:45,3rd,BSc Physics,3,3,2,12-Oct,3,3,3,...,4,4,4,3,4,Subscriptions,A feature to show what one should already know...,Decent,11.0,11-20
1,9/6/2026 15:00,3rd,BIT,4,4,2,20hrs,5,4,4,...,4,5,4,4,4,to pay,QNA,No,20.0,20+
2,9/6/2026 15:05,3rd,BIT,3,4,4,5,4,4,4,...,3,4,4,4,4,Giving short answers,Organizing my research,It helps in revising my study,5.0,6-10
3,9/6/2026 15:21,3rd,BIT,3,3,3,48 hrs,2,4,3,...,3,5,4,4,3,There unnecessary extra answers,Easier prompt to get enough required answer,No,48.0,20+
4,9/6/2026 16:22,3rd,BIT,3,3,1,12-Oct,2,3,2,...,2,3,2,2,2,Distraction,Idk,No,11.0,11-20


In [18]:
df.columns = df.columns.str.strip()

### Computing scores per respondents

In [22]:
df['behavioral_avg'] = df[['beh_daily_use', 'beh_time_spent', 'beh_attendance']].mean(axis = 1)
df['pu_avg'] = df[['pu_useful', 'pu_efficiency']].mean(axis=1)
df['interest_avg'] = df[['il_enjoyment', 'il_curiosity', 'em_connection']].mean(axis=1)
df['intention_avg'] = df[['biu_intention', 'biu_voluntary']].mean(axis=1)
df['srl_avg'] = df[['srl_strategy', 'srl_organize', 'srl_review', 'srl_persist', 'srl_evaluate']].mean(axis=1)

### Descriptive statistics of every new columns

In [26]:
constructs = ['behavioral_avg', 'pu_avg', 'interest_avg', 'intention_avg', 'srl_avg']

df[constructs].agg(['mean', 'std', 'min', 'max']).round(2)

,behavioral_avg,pu_avg,interest_avg,intention_avg,srl_avg
mean,2.98,3.47,3.46,3.66,3.61
std,0.91,0.96,0.82,1.19,0.78
min,1.67,2.00,1.67,1.00,2.00
max,5.00,5.00,5.00,5.00,4.60


##### Finding

Behavioural engagement is the weakest dimension (mean = 2.98), and it's dragged down by one item. "I regularly attend classes and actively participate" scored 2.35 by far the lowest item in the whole survey. Tool usage items scored fine (3.35, 3.24). So the gap isn't that students don't use tools; it's classroom participation.

### Ranking individual items to see the weak points

In [28]:
# sorting from lowest to highest
likert_items = ['beh_daily_use', 'beh_time_spent', 'beh_attendance',
                'pu_useful', 'pu_efficiency',
                'il_enjoyment', 'il_curiosity', 'em_connection',
                'biu_intention', 'biu_voluntary',
                'srl_strategy', 'srl_organize', 'srl_review', 'srl_persist', 'srl_evaluate']

df[likert_items].mean().sort_values().round(2)

beh_attendance    2.31
em_connection     3.06
beh_time_spent    3.25
pu_efficiency     3.31
beh_daily_use     3.38
srl_review        3.38
srl_persist       3.50
il_enjoyment      3.50
srl_strategy      3.56
pu_useful         3.62
biu_intention     3.62
srl_evaluate      3.69
biu_voluntary     3.69
il_curiosity      3.81
srl_organize      3.94
dtype: float64

#### Weakest items

- `beh_attendance` = 2.31, the lowest score in your entire survey. Students rate their class attendance/participation notably poorly, well below the neutral midpoint of 3 proving the finding mentioned above.

- `em_connection` = 3.06, barely above neutral. Students don't feel connected to instructors/peers through digital tools.

- `beh_time_spent` = 3.25, weakest engagement item after attendance.

#### Strongest items

- `srl_organize` = 3.94, students are best at organising their time/materials in advance. (strongest finding)

- `il_curiosity` = 3.81, students are curious about exploring new tool features.


### Checking correlation between constructs.

In [30]:
df[constructs].corr().round(2)

,behavioral_avg,pu_avg,interest_avg,intention_avg,srl_avg
behavioral_avg,1.00,0.74,0.68,0.58,0.70
pu_avg,0.74,1.00,0.69,0.59,0.69
interest_avg,0.68,0.69,1.00,0.78,0.72
intention_avg,0.58,0.59,0.78,1.00,0.91
srl_avg,0.70,0.69,0.72,0.91,1.00
